internal/service/source.go

```diff
@@ -188,3 +193,130 @@ func (h *ServiceHandler) UpdateSource(ctx context.Context, request server.Update
 
 	return server.UpdateSource200JSONResponse{}, nil
 }
+
+func (h *ServiceHandler) UploadRvtoolsFile(ctx context.Context, request server.UploadRvtoolsFileRequestObject) (server.UploadRvtoolsFileResponseObject, error) {
+    ctx, err := h.store.NewTransactionContext(ctx)
+    if err != nil {
+        return server.UploadRvtoolsFile500JSONResponse{}, nil
+    }
+
+    source, err := h.store.Source().Get(ctx, request.Id)
+    if err != nil {
+        _, _ = store.Rollback(ctx)
+        return server.UploadRvtoolsFile400JSONResponse{}, nil
+    }
+
+    if source == nil {
+        _, _ = store.Rollback(ctx)
+        return server.UploadRvtoolsFile404JSONResponse{}, nil
+    }
+
+    user := auth.MustHaveUser(ctx)
+    if source.OrgID != user.Organization {
+        _, _ = store.Rollback(ctx)
+        return server.UploadRvtoolsFile403JSONResponse{}, nil
+    }
+
+    var rvtoolsAgent *model.Agent
+    for _, a := range source.Agents {
+        rvtoolsAgent = &a
+        break
+    }
+
+    if rvtoolsAgent == nil {
+        newAgent := model.NewAgentForSource(uuid.New(), *source)
+        
+        if _, err := h.store.Agent().Create(ctx, newAgent); err != nil {
+            _, _ = store.Rollback(ctx)
+            return server.UploadRvtoolsFile500JSONResponse{}, nil
+        }
+        rvtoolsAgent = &newAgent
+    }
+
+    multipartReader := request.Body
+    
+    var rvtoolsContent []byte
+    
+    for {
+        part, err := multipartReader.NextPart()
+        if err == io.EOF {
+            break
+        }
+        if err != nil {
+            _, _ = store.Rollback(ctx)
+            return server.UploadRvtoolsFile400JSONResponse{}, nil
+        }
+        
+        if part.FormName() == "file" {
+            
+            rvtoolsContent, err = io.ReadAll(part)
+            if err != nil {
+                _, _ = store.Rollback(ctx)
+                return server.UploadRvtoolsFile400JSONResponse{}, nil
+            }
+            break
+        }
+    }
+    
+    if rvtoolsContent == nil {
+        _, _ = store.Rollback(ctx)
+        return server.UploadRvtoolsFile400JSONResponse{
+            Message: "No file was found in the request",
+        }, nil
+    }
+    
+    if len(rvtoolsContent) > 0 {
+        fmt.Printf("Received RVTools data with size: %d bytes\n", len(rvtoolsContent))
+        
+        if !rvtools.IsExcelFile(rvtoolsContent) {
+            _, _ = store.Rollback(ctx)
+            return server.UploadRvtoolsFile400JSONResponse{
+                Message: "The uploaded file is not a valid Excel (.xlsx) file. Please upload an RVTools export in Excel format.",
+            }, nil
+        }
+        
+        inventory, err := rvtools.ParseRVTools(rvtoolsContent)
```

Comment:
I would parse the inventory before getting the source because the vCenterID must match the on in the db.
It can happen the user used a real agent to get the first inventory and then it uploads the rvtools excel file. 
In this case, the old_vCenterID should be the same as the one from rvtools.